src/classes/SftpSessionHandler.ts

```diff
@@ -893,40 +866,38 @@ export class SftpSessionHandler {
           case 'xa+': // append and read (file must not exist)
           case 'a': // append
           {
-            tmp.file((err, name, fd, removeCallback) => {
-              if (err) {
+            const writeFile = new TemporaryFile(filePath);
```

Comment:
This could throw an error (if S3 settings have not been provided) -- it would be good to catch that error here.